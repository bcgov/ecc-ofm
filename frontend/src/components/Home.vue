<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <v-card
    title="OFM"
    subtitle="Temp Home Screen"
    text="..."
    style="height:100%;"
  ></v-card>
</template>
