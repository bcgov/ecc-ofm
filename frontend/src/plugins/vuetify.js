import '@fortawesome/fontawesome-free/css/all.css';
import '@mdi/font/css/materialdesignicons.css';
import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import { Ripple } from 'vuetify/lib/directives';

Vue.use(Vuetify, {
  directives: {
    Ripple
  }
});

export default new Vuetify({
  theme: {
    light: true,
    dark: false
  },
  icons: {
    iconfont: 'fa',
    values: {
      login: 'fas fa-user-clock',
      fast: 'fas fa-shipping-fast',
      sign_in: 'fas fa-sign-in-alt',
      info1: 'fas fa-info-circle',
      downArrow: 'fas fa-angle-down',
      upArrow: 'fas fa-angle-up',
      user: 'far fa-user',
      copy: 'fas fa-copy',
      search: 'fas fa-search',
      error: 'fas fa-exclamation-triangle',
      lock: 'fas fa-lock',
      info2: 'fas fa-info-circle fa-10x',
      question: 'fas fa-question-circle fa-10x',
      plus: 'fas fa-plus-circle',
      minus: 'fas fa-minus-circle',
    }
  }
});
