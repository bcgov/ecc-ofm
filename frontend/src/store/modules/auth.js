import ApiService from '@/common/apiService';
/* TODO: uncomment during integration with authservice...
import AuthService from '@/common/authService';
*/
import { Routes } from '@/utils/constants';
import {defineStore} from 'pinia';

export const authStore = defineStore('auth', {
  namespaced: true,
  state: () => ({
    acronyms: [],
    isAuthenticated: true  /* TODO: during integration with backend, replace true with... localStorage.getItem('jwtToken') !== null */,
    isAuthorizedUser: true /* TODO: during integration with backend, replace true with... localStorage.getItem('isAuthorizedUser') !== null */,
    userInfo: false,
    isValidChildCareProviderUser: localStorage.getItem('iisValidChildCareProviderUser') !== null,
    isValdProgramUser: localStorage.getItem('isValdProgramUser') !== null,
    isValidFinancialOpsUser: localStorage.getItem('isValidFinancialOpsUser') !== null,    
  }),
  getters: {
    acronymsGet: state => state.acronyms,
    isAuthenticatedGet: state => state.isAuthenticated,
    jwtTokenGet: () => localStorage.getItem('jwtToken'),
    userInfoGet: state => state.userInfo,
    //TODO: 3 temp roles ('CCP_ROLE', 'OPS_ROLE', 'PCM_ROLE') were created in auth.js (loosely
    //based on OFM requirements) for the purpose of achieving a 1st draft of the frontend that
    //will render a home screen and menu with minimal errors given no authorization/backend integration.
    //Thus the following related code is only temporarly and expected to be replaced...
    CCP_ROLE: state => state.isValidChildCareProviderUser,
    OPS_ROLE: state => state.isValdProgramUser,
    PCM_ROLE: state => state.isValidFinancialOpsUser
  },
  actions: {
    //sets Json web token and determines whether user is authenticated
    async setJwtToken(token = null) {
      if (token) {
        this.isAuthenticated = true;
        localStorage.setItem('jwtToken', token);
      } else {
        this.isAuthenticated = false;
        localStorage.removeItem('jwtToken');
      }
    },
    async setAuthorizedUser(isAdminUser){
      if (isAdminUser) {
        this.isAuthorizedUser = true;
        localStorage.setItem('isAuthorizedUser', 'true');
      } else {
        this.isAuthorizedUser = false;
        localStorage.removeItem(('isAuthorizedUser'));
      }
    },
    async setAuthorizedWebsocketUser(isAuthorizedWebsocketUser){
      if (isAuthorizedWebsocketUser) {
        this.isAuthorizedWebsocketUser = true;
        localStorage.setItem('isAuthorizedWebsocketUser', 'true');
      } else {
        this.isAuthorizedWebsocketUser = false;
        localStorage.removeItem(('isAuthorizedWebsocketUser'));
      }
    },
    async setUserInfo(userInf) {
      if (userInf) {
        this.userInfo = userInf;
      } else {
        this.userInfo = null;
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setChildCareProviderUser(isValidChildCareProviderUser) {
      if (isValidChildCareProviderUser) {
        this.isValidChildCareProviderUser = true;
        localStorage.setItem('isValidChildCareProviderUser', 'true');
      } else {
        this.isValidChildCareProviderUser = false;
        localStorage.removeItem(('isValidChildCareProviderUser'));
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setProgramUser(isValdProgramUser) {
      if (isValdProgramUser) {
        this.isValdProgramUser = true;
        localStorage.setItem('isValdProgramUser', 'true');
      } else {
        this.isValdProgramUser = false;
        localStorage.removeItem(('isValdProgramUser'));
      }
    },
      // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setFinancialOpsUser(isValidFinancialOpsUser) {
      if (isValidFinancialOpsUser) {
        this.isValidFinancialOpsUser = true;
        localStorage.setItem('isValidFinancialOpsUser', 'true');
      } else {
        this.isValidFinancialOpsUser = false;
        localStorage.removeItem(('isValidFinancialOpsUser'));
      }
    },
    //sets the token required for refreshing expired json web tokens
    async logoutState() {
      localStorage.removeItem('jwtToken');
      this.userInfo = false;
      this.isAuthenticated = false;
    },
    async getUserInfo() {
      const token = localStorage.getItem('jwtToken');

      if(!token){
        await this.getJwtToken();
      }

      if(localStorage.getItem('jwtToken')) {
        const response = await ApiService.apiAxios.get(Routes.USER);
        await this.setUserInfo(response.data);
      }
    },
    //retrieves the json web token from local storage. If not in local storage, retrieves it from API
    async getJwtToken() {
      try {
        const token = localStorage.getItem('jwtToken');
        if (this.isAuthenticated && !!token) {
          /* TODO: uncomment during integration with authservice/backend...
          const response = await AuthService.refreshAuthToken(token);
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        } else {
          /* TODO: uncomment during integration with backend...
          const response = await AuthService.getAuthToken();
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        }
      } catch (e) {
        // Remove tokens from localStorage and update state
        await this.setJwtToken(null);
        throw e;
      }
    },
    //TODO: uncomment 'response' below during integration with authorization/backend...
    async setAuthorizations(/*response*/) {
    /* TODO: uncomment during integration with backend...
      if (response.jwtFrontend) {
        await this.setJwtToken(response.jwtFrontend);
      }

      ApiService.setAuthHeader(response.jwtFrontend);
    */
      await this.setAuthorizedUser(true);
      await this.setChildCareProviderUser(true);
      await this.setProgramUser(true);
      await this.setFinancialOpsUser(true);
    }
  }
});
