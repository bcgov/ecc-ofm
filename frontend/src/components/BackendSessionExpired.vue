<template>
  <div style="display: none">
    <a id="logout_href" :href="link" />
  </div>
</template>

<script>
import { Routes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'
import { mapActions } from 'pinia'
export default {
  name: 'BackendSessionExpired',
  data() {
    return {
      link: Routes.SESSION_EXPIRED
    }
  },
  mounted() {
    window.location = document.getElementById('logout_href').href
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    logout() {
      this.setJwtToken()
    }
  }
}
</script>
