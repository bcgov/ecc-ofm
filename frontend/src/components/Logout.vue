<template>
  <v-container fluid class="full-height">
    <!-- login article -->
    <article name="logout-banner">
      <v-row align="center" justify="center">
        <v-col xs="10" sm="10" md="8" lg="4" xl="3">
          <v-card class="session-expired-card">
            <v-card-title class="gov-header">
              <h4 id="logout_text">Logged Out</h4>
            </v-card-title>
            <v-card-text id="logout_descriptor">
              <v-row style="margin: 0.3rem"> You have Logged out. </v-row>
              <a
                id="login-button"
                :href="routes.LOGIN"
                class="ma-1"
                dark
                color="#003366"
                @click="clearStorage"
                >Log In</a
              ><span>again if you wish to continue.</span>
            </v-card-text>
          </v-card>
        </v-col>
      </v-row>
    </article>
  </v-container>
</template>

<script>
import { mapActions } from 'pinia'
import { Routes } from '../utils/constants'
import { useAuthStore } from '@/stores/auth'

export default {
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Logout',

  data() {
    return {
      routes: Routes
    }
  },
  created() {
    this.setJwtToken()
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    clearStorage() {
      this.setJwtToken()
    }
  }
}
</script>

<style scoped>
.full-height {
  height: 100%;
}
.session-expired-card {
  margin-top: 15rem;
  width: 100%;
  background: #d9e7d8;
}
</style>
