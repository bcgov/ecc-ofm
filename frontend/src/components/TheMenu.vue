<template>
  <div>
    <AppMenuItem icon="mdi-home-outline" :to="{ name: 'home' }">Home</AppMenuItem>
    <AppMenuItem :to="{ name: 'messaging' }">
      <div class="badge-wrapper">
        <v-badge v-if="unreadNotificationCount > 0" :content="unreadNotificationCount" color="red" offset-x="18"
          offset-y="17">
          <v-icon class="badge-icon" aria-hidden="false" icon="mdi-email-outline" size="30" />
        </v-badge>
        <v-icon v-else-if="!unreadNotificationCount" class="badge-icon" aria-hidden="false" icon="mdi-email-outline"
          size="30" />
        Messaging
      </div>
    </AppMenuItem>
    <AppMenuItem icon="mdi-text-box-outline">Reporting</AppMenuItem>
    <AppMenuItem icon="mdi-currency-usd">Funding</AppMenuItem>
    <AppMenuItem icon="mdi-folder-outline">Documents</AppMenuItem>
    <AppMenuItem icon="mdi-file-document-edit-outline">Applications</AppMenuItem>
    <AppMenuItem icon="mdi-help" :to="{ name: 'resources' }">Resources</AppMenuItem>
    <AppMenuItem icon="mdi-cog-outline">Settings</AppMenuItem>
  </div>
</template>

<script>
import { mapState, mapActions } from 'pinia'
import { useAppStore } from '@/stores/app'
import { useAuthStore } from '@/stores/auth'
import { useNotificationsStore } from '@/stores/notifications'
import AppMenuItem from '@/components/ui/AppMenuItem.vue'

export default {
  components: { AppMenuItem },
  computed: {
    ...mapState(useAppStore, ['showMenu']),
    ...mapState(useAuthStore, ['userInfo', 'isAuthenticated']),
    ...mapState(useNotificationsStore, ['unreadNotificationCount']),
  },
  async created() {
    try {
      await this.getNotifications(this.userInfo.contactId)
    } catch (error) {
      console.info(error)
    }
  },
  methods: {
    ...mapActions(useNotificationsStore, ['getNotifications']),
  },
}
</script>

<style scoped>
.badge-wrapper {
  display: flex;
  align-items: center;
  width: 100%;
  height: 100%;
}

.badge-icon {
  color: #003366;
  margin-right: 15px;
}
</style>
