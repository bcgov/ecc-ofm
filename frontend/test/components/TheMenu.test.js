import { describe, test } from 'vitest'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import TheMenu from '@/components/TheMenu.vue'
import { useAuthStore } from '@/stores/auth.js'
import { createTestingPinia } from '@pinia/testing'
import { mount } from '@vue/test-utils'

const vuetify = createVuetify({ components, directives })

describe('TheMenu', () => {
  const global = { global: { plugins: [createTestingPinia(), vuetify] } }
  const authStore = useAuthStore()
  authStore.isAuthenticated = true
  authStore.currentFacility = { facilityName: 'UT Facility' }

  // Tests that all Menu Items display for a user with all roles
  test('all menu items visible', async () => {
    authStore.userInfo = {
      organizationName: 'UT Org',
      role: 2,
    }
    const wrapper = mount(TheMenu, global)
    expect(wrapper.text()).toContain('Home')
    expect(wrapper.text()).toContain('Messaging')
    expect(wrapper.text()).toContain('Reporting')
    expect(wrapper.text()).toContain('Funding')
    expect(wrapper.text()).toContain('Documents')
    expect(wrapper.text()).toContain('Applications')
    expect(wrapper.text()).toContain('Resources')
    expect(wrapper.text()).toContain('Settings')
    expect(wrapper.html()).toMatchSnapshot()
  })

  // Tests that the Settings Menu Item isn't displayed
  // for a user without 2 = 'Account Management'
  test('Settings menu item hidden', async () => {
    authStore.userInfo = {
      organizationName: 'UT Org',
      role: 1,
    }
    const wrapper = mount(TheMenu, global)
    expect(wrapper.text()).toContain('Home')
    expect(wrapper.text()).toContain('Messaging')
    expect(wrapper.text()).toContain('Reporting')
    expect(wrapper.text()).toContain('Funding')
    expect(wrapper.text()).toContain('Documents')
    expect(wrapper.text()).toContain('Applications')
    expect(wrapper.text()).toContain('Resources')
    expect(wrapper.text()).not.toContain('Settings')
  })
})
