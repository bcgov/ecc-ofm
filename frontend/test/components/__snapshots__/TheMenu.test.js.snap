// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheMenu > all menu items visible 1`] = `
"<div data-v-ca5eac82=\\"\\"><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-home-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Home
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!---->
      <!--v-if-->
      <div data-v-ca5eac82=\\"\\" class=\\"badge-wrapper\\"><i data-v-ca5eac82=\\"\\" class=\\"mdi-email-outline mdi v-icon notranslate v-theme--light badge-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"false\\"></i> Messaging </div>
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-text-box-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Reporting
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-currency-usd mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Funding
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-folder-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Documents
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-file-document-edit-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Applications
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-help mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Resources
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-cog-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Settings
    </div>
    <!---->
  </a></div>"
`;

exports[`TheMenu > allMenuItems 1`] = `
"<div data-v-ca5eac82=\\"\\"><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-home-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Home
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!---->
      <!--v-if-->
      <div data-v-ca5eac82=\\"\\" class=\\"badge-wrapper\\"><i data-v-ca5eac82=\\"\\" class=\\"mdi-email-outline mdi v-icon notranslate v-theme--light badge-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"false\\"></i> Messaging </div>
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-text-box-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Reporting
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-currency-usd mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Funding
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-folder-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Documents
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-file-document-edit-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Applications
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-help mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Resources
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-cog-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Settings
    </div>
    <!---->
  </a></div>"
`;

exports[`TheMenu > allMenuItems 2`] = `
"<div data-v-ca5eac82=\\"\\"><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-home-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Home
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!---->
      <!--v-if-->
      <div data-v-ca5eac82=\\"\\" class=\\"badge-wrapper\\"><i data-v-ca5eac82=\\"\\" class=\\"mdi-email-outline mdi v-icon notranslate v-theme--light badge-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"false\\"></i> Messaging </div>
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-text-box-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Reporting
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-currency-usd mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Funding
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-folder-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Documents
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-file-document-edit-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Applications
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-help mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Resources
    </div>
    <!---->
  </a><a data-v-782d38a9=\\"\\" data-v-ca5eac82=\\"\\" class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--variant-text menu-item\\" style=\\"color: #003366; caret-color: #003366;\\" tabindex=\\"0\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
    <!---->
    <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
      <!---->
      <!----><i data-v-782d38a9=\\"\\" class=\\"mdi-cog-outline mdi v-icon notranslate v-theme--light menu-icon\\" style=\\"font-size: 30px; height: 30px; width: 30px;\\" aria-hidden=\\"true\\"></i>Settings
    </div>
    <!---->
  </a></div>"
`;
